### DISCUSSION TITLE
Access the statistics data for a specified time range

### DISCUSSION THREAD

**Discussion by Fredo70** [custom-components/pyscript#530] [Q&A] [answered]
Hi.
I would like to develop a service that does the following tasks:
- Query statistics data from multiple sensors (typically energy) for a given time range.
- Prepare this data as JSON
- Send the JSON payload to a reporting service via REST API.
- The reporting service responds with a PDF file in the content.
- The PDF file is sent by email to a specified address.

The problem is, I am neither a HA nor a python specialist ;-) But I already have some basic knowledge.
I've studied the developer documentation of HA and I think developing my own integration seems too complicated for me. And pyscript would probably be ideal to start with.
So the service will have lots of configuration parameters.
REST call and send email are (I think) no problem.
But before I start, I wonder how do I get access to the stats? I could query directly into the SQLite database. But aren't there any functions from HA that I can use? I haven't found that anywhere. Is there anything like that?
And is my plan feasible?
Thanks a lot

**Comment by ALERTua** [comment, marked as answer]
I guess you could even skip the JSON and REST API and reporting service and go straight to PDF via pyscript.
Historical data was discussed here #405 #268 with working examples.

**Reply by Fredo70** [reply]
Thank you. It gets me further.
The reporting service is much easier for me (I already know it). And it has enormous flexibility.
